core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdoc_conv.go

```diff
@@ -66,58 +67,66 @@ func (v jsonValue) toBytes() ([]byte, error) {
 }
 
 func couchDocToKeyValue(doc *couchDoc) (*keyValue, error) {
-	// initialize the return value
-	var returnValue []byte
-	var err error
-	// create a generic map unmarshal the json
-	jsonResult := make(map[string]interface{})
-	decoder := json.NewDecoder(bytes.NewBuffer(doc.jsonValue))
-	decoder.UseNumber()
-	if err = decoder.Decode(&jsonResult); err != nil {
+	docFields, err := validateAndRetrieveFields(doc)
+	if err != nil {
 		return nil, err
 	}
-	// verify the version field exists
-	if _, fieldFound := jsonResult[versionField]; !fieldFound {
-		return nil, errors.Errorf("version field %s was not found", versionField)
+	version, metadata, err := decodeVersionAndMetadata(string(docFields.versionAndMetadata))
+	if err != nil {
+		return nil, err
 	}
-	key := jsonResult[idField].(string)
-	// create the return version from the version field in the JSON
+	return &keyValue{
+		doc.id, docFields.revision,
+		&statedb.VersionedValue{
+			Value:    docFields.value,
+			Version:  version,
+			Metadata: metadata,
+		},
+	}, nil
+}
 
-	returnVersion, returnMetadata, err := decodeVersionAndMetadata(jsonResult[versionField].(string))
-	if err != nil {
+type couchDocFields struct {
+	revision           string
+	value              []byte
+	versionAndMetadata []byte
+}
+
+func validateAndRetrieveFields(doc *couchDoc) (*couchDocFields, error) {
+	jsonDoc := make(jsonValue)
+	decoder := json.NewDecoder(bytes.NewBuffer(doc.jsonValue))
+	decoder.UseNumber()
+	if err := decoder.Decode(&jsonDoc); err != nil {
 		return nil, err
 	}
-	var revision string
-	if jsonResult[revField] != nil {
-		revision = jsonResult[revField].(string)
+
+	docFields := &couchDocFields{}
+	if jsonDoc[revField] != nil {
+		docFields.revision = jsonDoc[revField].(string)
+	}
+	if jsonDoc[versionField] == nil {
+		return nil, fmt.Errorf("version field %s was not found", versionField)
 	}
+	docFields.versionAndMetadata = []byte(jsonDoc[versionField].(string))
```

Comment:
`versionAndMetadata` string might break the code consistency because in a struct, we have always used `versionAndMetadata []byte` to store the version and metadata. Moreover, we can see if it is possible to remove the string representation altogether but not a task of this PR. Hence, I leave it as is for now.

Reply:
> `versionAndMetadata string` might break the code consistency because in a struct, we have always used `versionAndMetadata []byte` to store the version and metadata.

Not sure that I understand this. The only struct I could find was the one that you introduced in this PR only. The lowest level functions in current [code](https://github.com/hyperledger/fabric/blob/37fc7720f139ad2d383c63ffce14d8f9b2469abb/core/ledger/kvledger/txmgmt/statedb/statecouchdb/version_metadata_encoding.go#L17) treat this as string type. So, I pointed this inconsistency rather.

Reply:
Oh, yes. Only delete() of other fields are present not this one. Let me see why I added this and remove it.

Reply:
Done. It was introduced by mistake.